scripts/OutputDifferences.Rmd

```diff
@@ -0,0 +1,129 @@
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+## Differences in Hector outputs
+
+Hello, this is `leeyabot`! `r emo::ji("robot")`
+
+The current pull request's outputs differ from `master` as follows:  
+
+
+``` {r differences, echo = FALSE, message = FALSE}
+# Load packages
+library(dplyr)
+library(ggplot2)
+library(hector)
+library(emo)
+library(here)
+
+# Set root directory
+BASE_DIR <- here::here()
+
+# Read in comp-data
+path <- file.path(BASE_DIR, "tests", "testthat", "compdata", "hector_comp.csv")
+comp_data <- read.csv(path)
+
+# Variables of interest
+vars <- c(ATMOSPHERIC_C(), GLOBAL_TEMP(), RF_TOTAL())
+
+# Organize data
+compdata <- comp_data %>% 
+    select(scenario, year, variable, value, units, version) %>%
+    filter(scenario == "hector_rcp45.ini") %>%
+    filter(variable %in% vars) %>%
+    mutate(scenario = "rcp45") %>%
+    arrange(variable)
+    
+# Read in files for just RCP 4.5 - v3
+rcp45 <- file.path(BASE_DIR, "inst", "input", "hector_rcp45.ini")
+
+# Run core and retrieve variable
+core <- newcore(rcp45)
+invisible(run(core))
+output <- fetchvars(core, core$strtdate:core$enddate, vars, "rcp45")
+output <- output %>% 
+    mutate(version = "latest") %>%
+    arrange(variable)
+
+# Find differences between versions
+differences <- compdata %>% mutate(diff = output$value - compdata$value)
+differences <- differences %>% select(scenario, year, variable, units, version, diff)
+```
+
+
+```{r summary-table, echo= FALSE}
+
+# Organize data
+output %>%
+    select(latest = value) -> new_values
+
+both_versions <- compdata %>%
+    select(year, variable, value) %>%
+    rename(v3.0.0 = value) %>%
```

Comment:
Instead of naming v3.0.0 what about naming it something like `compdata_value` because `leeyabot` will not always be comparing data to v3.0.0 in the near future, it will be using Hector v3.5.0. Making that column name slightly more generic would help avoid confusion as Hector continues to develop.